can/broadcastmanager.py

```diff
@@ -71,44 +102,55 @@ def start(self):
 class ModifiableCyclicTaskABC(CyclicSendTaskABC):
     """Adds support for modifying a periodic message"""
 
-    def modify_data(self, message):
-        """Update the contents of this periodically sent message without altering
-        the timing.
+    def modify_data(self, messages):
+        """Update the contents of the periodically sent messages, without
+        altering the timing.
+
+        :param Union[List[can.Message], tuple(can.Message), can.Message] messages:
+            The messages with the new :attr:`can.Message.data`.
+
+            Note: The arbitration ID cannot be changed.
 
-        :param can.Message message:
-          The message with the new :attr:`can.Message.data`.
-          Note: The arbitration ID cannot be changed.
+            Note: The number of new cyclic messages to be sent must be equal
+            to the original number of messages originally specified for this
+            task.
         """
-        self.message = message
+        messages = self._check_and_convert_messages(messages)
```

Comment:
Since we are changing the code here, maybe we should enforce the fact that the arbitration ID cannot be changed.